libs/server/Storage/Session/ObjectStore/HashOps.cs

```diff
@@ -537,5 +538,98 @@ public GarnetStatus HashIncrement<TObjectContext>(byte[] key, ArgSlice input, ou
+        }
+
+        /// <summary>
+        /// Returns the time-to-live (TTL) of a hash key.
+        /// </summary>
+        /// <typeparam name="TObjectContext">The type of the object context.</typeparam>
+        /// <param name="key">The key of the hash.</param>
+        /// <param name="isMilliseconds">Indicates whether the TTL is in milliseconds.</param>
+        /// <param name="isTimestamp">Indicates whether the TTL is a timestamp.</param>
+        /// <param name="input">The input object containing the operation details.</param>
+        /// <param name="outputFooter">The output footer object to store the result.</param>
+        /// <param name="objectContext">The object context for the operation.</param>
+        /// <returns>The status of the operation.</returns>
+        public GarnetStatus HashTimeToLive<TObjectContext>(ArgSlice key, bool isMilliseconds, bool isTimestamp, ref ObjectInput input, ref GarnetObjectStoreOutput outputFooter, ref TObjectContext objectContext)
+            where TObjectContext : ITsavoriteContext<byte[], IGarnetObject, ObjectInput, GarnetObjectStoreOutput, long, ObjectSessionFunctions, ObjectStoreFunctions, ObjectStoreAllocator>
+        {
+            var innerInput = new ObjectInput(input.header, ref input.parseState, arg1: isMilliseconds ? 1 : 0, arg2: isTimestamp ? 1 : 0);
+
+            return ReadObjectStoreOperationWithOutput(key.ToArray(), ref innerInput, ref objectContext, ref outputFooter);
+        }
+
+        /// <summary>
+        /// Removes the expiration time from a hash key, making it persistent.
+        /// </summary>
+        /// <typeparam name="TObjectContext">The type of the object context.</typeparam>
+        /// <param name="key">The key of the hash.</param>
+        /// <param name="input">The input object containing the operation details.</param>
+        /// <param name="outputFooter">The output footer object to store the result.</param>
+        /// <param name="objectContext">The object context for the operation.</param>
+        /// <returns>The status of the operation.</returns>
+        public GarnetStatus HashPersist<TObjectContext>(ArgSlice key, ref ObjectInput input, ref GarnetObjectStoreOutput outputFooter, ref TObjectContext objectContext)
+            where TObjectContext : ITsavoriteContext<byte[], IGarnetObject, ObjectInput, GarnetObjectStoreOutput, long, ObjectSessionFunctions, ObjectStoreFunctions, ObjectStoreAllocator>
+            => RMWObjectStoreOperationWithOutput(key.ToArray(), ref input, ref objectContext, ref outputFooter);
+
+        public GarnetStatus HashCollect<TObjectContext>(ArgSlice key, ref ObjectInput input, ref TObjectContext objectContext)
+            where TObjectContext : ITsavoriteContext<byte[], IGarnetObject, ObjectInput, GarnetObjectStoreOutput, long, ObjectSessionFunctions, ObjectStoreFunctions, ObjectStoreAllocator>
+        {
+            if (key.ReadOnlySpan.SequenceEqual("*"u8))
```

Comment:
This should be really happening under a lock (see SingleWriterMultiReaderLock) similar to how a checkpoint is executed. This is to avoid overwhelming the server with many parallel scans if the user calls HCOLLECT in parallel.

Reply:
If a `HCOLLECT *` is called while a scan is running (either background or user-initiated), then it should return ERR HCOLLECT scan already in progress. See how we do this for `SAVE` for example.

Reply:
Done